klog.go

```diff
@@ -802,10 +803,16 @@ func (l *loggingT) infoS(logger *logr.Logger, filter LogFilter, depth int, msg s
 // set log severity by s
 func (l *loggingT) printS(err error, s severity, depth int, msg string, keysAndValues ...interface{}) {
 	b := &bytes.Buffer{}
-	b.WriteString(fmt.Sprintf("%q", msg))
+	// The message is never quoted. New lines get indented.
+	writeString(b, msg)
+	// Enhance readability by inserting : between message and key/value
+	// pairs, but only when the message does not already end with
+	// punctation.
+	if len(msg) > 0 && !unicode.IsPunct(rune(msg[len(msg)-1])) {
+		b.WriteString(":")
```

Comment:
Can we think about using some other separator here? I've seen many logs use something like `klog.InfoS("Warning: some text", key, value)`

Reply:
Perhaps ` | `, with a space before and after it?